### PR TITLE
Rules: access to local objects

### PR DESCRIPTION
**Pull request by leogermond**
Local scopes are subprocedure and task bodies.

### CODE REVIEW

File: lkql_checker/share/lkql/access_to_local_objects.lkql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+fun local_scope(e) =
+    match e
```
Comment (ArnaudCharlet) [inline lkql_checker/share/lkql/access_to_local_objects.lkql:2]: I think this is still missing several cases of local variables such as declare blocks (which can also occur in package body elaboration), protected entries. Does SubpBody include protected subprogram bodies? If not, these need to be also included.

Comment (leogermond) [reply lkql_checker/share/lkql/access_to_local_objects.lkql:2]: Is it possible to have declare blocks that are not enclosed in a subprogram or any such construct? Anyway I added them as type of local scope, this seems like a practical function to have laying around.

Comment (ArnaudCharlet) [reply lkql_checker/share/lkql/access_to_local_objects.lkql:2]: Yes, for package body elaboration code as I mentioned:
package body xx is
...
begin
   <put any Ada code here, including a declare block>
end xx;

**Review by ArnaudCharlet — DISMISSED** [review]

**Comment by leogermond** [comment]
I've rebased instead of merging so the older commits and review are lost :facepalm:

**Review by ArnaudCharlet — APPROVED** [review]

**Pull request merged**